idseq_dag/util/s3.py

```diff
@@ -215,6 +215,18 @@ def install_s3mi(installed={}, mutex=TraceLock("install_s3mi", multiprocessing.R
             installed['time'] = time.time()
 
 
+def install_s3parcp(installed={}, mutex=TraceLock("install_s3parcp", multiprocessing.RLock())):  # pylint: disable=dangerous-default-value
```

Comment:
Originally I was going to add installation to the dockerfile. However, during index generation we are not running the pipeline in docker. s3mi is also installed this way so I created a similar approach for s3parcp. Is there a reason we are not using docker to run the index generation? I would prefer all dependencies to be handled through docker so we can safely assume that all dependencies in the Dockerfile will be available at runtime. Would like some feedback on this.

Reply:
> Is there a reason we are not using docker to run the index generation? 

I had the same question myself. I don't think there is a good reason. I would also prefer our dependencies to be handled in a standard way. That said, I think the standard for index generation is to install them in the AMI cloud init script, which runs when the EC2 instance initializes. Any reason you can't do that here? Then you wouldn't need to do the wacky checking for previous installations. 

s3mi is installed in the cloud init script. See terraform/envs/staging/index-rapsearch/templates/cloud_init.sh.tpl .

Reply:
I will add it to there and the dockerfile, I agree that is better. The way we are doing the index generation seems unusual to me. A lambda to manage an autoscaling group with an init template that does the whole job. To me this seems like it could be an ECS task. Lambda isn't the best fit because of all the language-external dependencies.